app/models.py

```diff
@@ -0,0 +1,607 @@
+        """
+        if User.get_by_username(username):
+            raise ValidationException("The username '{}' already exist"
+                                      .format(username))
+
+    @staticmethod
+    def check_email_uniqueness(email):
+        """
+       checks the database if the email passed in as argument 
+       doesn't exist within the database
+       :param email: 
+       :return boolean (True if the username name doesn't exist else False): 
+       """
+        if User.get_by_email(email):
+            raise ValidationException("The email '{}' already exist"
+                                      .format(email))
+
+    def delete(self):
+        """deletes a user from the database"""
+        db.session.delete(self)
+        db.session.commit()
+
+
+class ShortenUrl(db.Model):
+    """
+        This class defines the shorten_url table schema and also handles
+        shorten_url object  related functionalities.
+    """
+    __table_name__ = "shorten_urls"
+    id = db.Column(db.Integer, primary_key=True)
+    shorten_url_name = db.Column(db.String(20), unique=True, index=True,
+                                 nullable=False)
+    date_added = db.Column(db.DateTime, default=datetime.utcnow())
+    user = db.Column(db.Integer, db.ForeignKey('users.id'))
+    long_url = db.Column(db.Integer, db.ForeignKey('url.id'))
+    is_active = db.Column(db.Boolean, default=True)
+    deleted = db.Column(db.Boolean, default=False)
+    visits = db.relationship("ShortenUrlVisitLogs", backref='shorten_urls',
+                             lazy='dynamic')
+
+    @property
+    def name(self):
+        """ 
+        returns the value of the shorten_url_name attribute concatenated
+        with the url_root of the request
+        """
+        return request.url_root + self.shorten_url_name
+
+    @name.setter
+    def name(self, new_name):
+        """
+        sets the shorten_url_name attribute value to the argument passed 
+        into the function
+        :param new_name: 
+        """
+        self.shorten_url_name = new_name
+
+    @property
+    def visit(self):
+        """ 
+        returns an sqlalchemy append base query for all
+        the details of visits made to the shorten url
+        """
+        return self.visits
+
+    @staticmethod
+    def get_short_url_by_name(name):
+        """
+        returns a shorten url based on the name passed 
+        in as argument
+        :param name: 
+        :return ShortUrl Object: 
+        """
+        return ShortenUrl.query.filter_by(shorten_url_name=name).first()
+
+    @staticmethod
+    def save(shorten_url_name, user, long_url):
+        """
+        creates a ShortenUrl object and saves it to the database
+        :param shorten_url_name: 
+        :param user: 
+        :param long_url: 
+        """
+        shorten_url = ShortenUrl(shorten_url_name=shorten_url_name,
+                                 user=user.id, long_url=long_url.id)
```

Comment:
That line of code actually creates a new instance a shorten_url which requires all the fields to be provided.